### PR TITLE
I-#47: Adding branching related guidelines to doc site

### LINKED ISSUE #47: Git branch management documentation
As a BloodConnect developer
Given BloodConnect is using git as VCS
When I want to manage my code using git
Then I need a documentation to follow

Git branch management documentation must cover
1. Main branch name
2. Branch naming convention
3. Branch lifecycle with proper image
4. Release management by git tag

Acceptance Criteria
1. Docs must be written using restructuredText (reST)
2. Docs must be in docs folder
3. Docs must have properly linked to developer documentation
4. After merging the PR docs must have properly visible to github pages

### PR DESCRIPTION
**Pull request by shuaibq3**
Fixes #47

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [ ] New feature
- [ ] Refactor
- [ ] Infrastructure
- [x] Documentation

## How would you describe the changes in a changelog (1-2 lines)?
Adding branch management related guidelines to our doc site.

Provide a brief overview of the changes introduced in this pull request.
Added a `rst` file to `/docs/development` named `BranchManagement.rst`, that contains some guidelines on how branches are managed in this repo, including branch naming convention.

## How to test

Build the doc site locally and verify that:
- A link to branch management is visible in `/development/index.html#guidelines`, below `Ticket Lifecycle` page.
- The contents of the page express the intent properly. Please comment if further explanation is needed.

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [x] I have added or updated necessary documentation to reflect the changes.
- [x] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [x] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: docs/development/BranchManagement.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Branch Naming Convention
+~~~~~~~~~~~~~~~~~~~~~~~~
+In case of working in a branch that will eventually end up as a PR, the branch name **must** contain a reference to the github issue number that it is implementing. The naming convention would be `I<issue_number>-<text reflecting the changes>`.
+
+For example, the github issue number of the requirement to add this page to our documentation site is 47. Following the above guideline, the branch that implements this change is named `I47-branch-mgt-doc`.
+
+Branch Lifecycle
+~~~~~~~~~~~~~~~~
+- As the PR branches are primarily intended to be merged to `master`, the feature branch should ideally be taken from the latest commit of `master` branch. That is where the branch starts in journey.
```
Comment (melon-ruet) [inline docs/development/BranchManagement.rst:17]: Want to add a picture like this?
https://nvie.com/posts/a-successful-git-branching-model/

File: docs/development/index.rst [hunk 2]
```diff
@@ -13,6 +13,7 @@ Guidelines
    CloudDevGuideline
    PRGuideline
    TicketLifecycle
+   BranchManagement
```
Comment (melon-ruet) [inline docs/development/index.rst:16]: I am thinking, do we want to merge PRGuideline and BranchManagement and create a new doc named "Git Management"?

File: docs/development/BranchManagement.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+- Once a PR is reviewed and merged, the branch should be deleted from github. It can be done from the PR page of github, or through git command.
```
Comment (melon-ruet) [inline docs/development/BranchManagement.rst:18]: Is there any commit guideline?

File: docs/development/BranchManagement.rst (hunk 1 above)
Comment (shuaibq3) [reply docs/development/BranchManagement.rst]: good point.. i was missing the bigger picture of this doc. added a git workflow diagram in commit [b5afa7f](https://github.com/CraftsmenLtd/BloodConnect/pull/111/commits/b5afa7f41064a3d53d68aa226af00a5354eb744e)

File: docs/development/index.rst (hunk 2 above)
Comment (shuaibq3) [reply docs/development/index.rst]: yes agreed. was missing the bigger picture here. merged git related docs in commit [b5afa7f](https://github.com/CraftsmenLtd/BloodConnect/pull/111/commits/b5afa7f41064a3d53d68aa226af00a5354eb744e)

File: docs/development/BranchManagement.rst (hunk 3 above)
Comment (shuaibq3) [reply docs/development/BranchManagement.rst]: yes there is. added it to the doc in commit [b5afa7f](https://github.com/CraftsmenLtd/BloodConnect/pull/111/commits/b5afa7f41064a3d53d68aa226af00a5354eb744e)

**Review by melon-ruet — APPROVED** [review]

**Pull request merged**